src/main/kotlin/com/sourcegraph/cody/history/HistoryService.kt

```diff
@@ -61,45 +57,51 @@ class HistoryService(private val project: Project) :
 
   @Synchronized
   fun updateDefaultContextState(contextState: EnhancedContextState) {
-    state.defaultEnhancedContext = EnhancedContextState()
-    state.defaultEnhancedContext?.copyFrom(contextState)
+    getOrCreateActiveAccountHistory().defaultEnhancedContext = EnhancedContextState()
```

Comment:
This PR is big enough already but we should consider renaming "History" to "State". It's a bit strange that `updateDefaultContextState` modifies the "history".